crates/orchestrator/src/database/mod.rs

```diff
@@ -33,6 +34,8 @@ pub trait Database: Send + Sync {
     ) -> Result<()>;
 
     async fn update_metadata(&self, job: &JobItem, metadata: HashMap<String, String>) -> Result<()>;
+    async fn get_latest_job_by_type(&self, job_type: JobType) -> Result<Option<JobItem>>;
+    async fn get_all_jobs(&self, job_type: JobType) -> Result<Cursor<JobItem>>;
```

Comment:
`Cursor` is a mongodb type and the DatabaseTrait is generic across the database you're using. So we should change this to `Vec` and handle the conversion for Cursor -> Vec inside the MongoDB function

Reply:
There is no need for this function for now as test is working without it. Will remove this and if in future we need to add this function will surely follow the above method.